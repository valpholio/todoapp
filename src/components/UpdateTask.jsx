import React, { useState } from 'react';
import PropTypes from 'prop-types';

const UpdateTask = ({createTask, exitAddTaskMode,editTask, exitEditTaskMode, ...task}) => {
    const [title, setTitle] = useState(task.title ||'');
    const [text, setText] = useState(task.text ||'');

    const createHandler = () => {
        createTask(title,text);
        exitAddTaskMode();
    };
    const editHandler = () => {
        editTask(task.id, title, text);
        exitEditTaskMode();
    };

    const handler = task.id? editHandler: createHandler;
    return (
        <div>
            <div>
            Title:<br/>
            <input type='text' value={title} onChange={e=>setTitle(e.target.value)} className='form-control' />
                <br/>
            Text:<br/>
            <textarea name='text' rows='5' value={text} onChange={e=>setText(e.target.value)} className='form-control' ></textarea>
            </div>
            <button onClick={handler} disabled={!(title && text)} className='btn btn-default' >Save</button>
            <button onClick={task.id? exitEditTaskMode: exitAddTaskMode} className='btn btn-default'>Cancel</button>
        </div>
    );
};

UpdateTask.propTypes = {
    createTask: PropTypes.func,
    exitAddTaskMode: PropTypes.func,
    editTask: PropTypes.func,
    exitEditTaskMode: PropTypes.func,
};

export default UpdateTask;