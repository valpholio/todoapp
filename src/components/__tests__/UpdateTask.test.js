import React from 'react';
import renderer from 'react-test-renderer';
import UpdateTask from '../UpdateTask.jsx';

describe('UpdateTask', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<UpdateTask/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});