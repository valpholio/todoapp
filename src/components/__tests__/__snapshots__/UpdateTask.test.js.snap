// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateTask snapshot renders 1`] = `
<div>
  <div>
    Title:
    <br />
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    Text:
    <br />
    <textarea
      className="form-control"
      name="text"
      onChange={[Function]}
      rows="5"
      value=""
    />
  </div>
  <button
    className="btn btn-default"
    disabled={true}
    onClick={[Function]}
  >
    Save
  </button>
  <button
    className="btn btn-default"
  >
    Cancel
  </button>
</div>
`;
